File: funcscript.test/ParseTests/NakedKeyValueCollectionWithImplicitReturnTest.cs
using System.Collections.Generic;
using FuncScript.Core;
using NUnit.Framework;

namespace FuncScript.Test;

public class NakedKeyValueCollectionWithImplicitReturnTest
{
    [Test]
    public void NakedKeyValueCollectionWithImplicitReturParsedCorrectly()
    {
        var expression = "x:12;x*2;";
        var syntaxErrors = new List<FuncScriptParser.SyntaxErrorData>();
        var context = new FuncScriptParser.ParseContext(new DefaultFsDataProvider(), expression, syntaxErrors);

        var (exp, parseNode, _) = FuncScriptParser.Parse(context);

        Assert.IsNotNull(parseNode);
        Assert.AreEqual(FuncScriptParser.ParseNodeType.KeyValueCollection, parseNode.NodeType);
        Assert.That(parseNode.Children.Count, Is.EqualTo(2));

        var keyValuePairNode = parseNode.Children[0];
        Assert.AreEqual(FuncScriptParser.ParseNodeType.KeyValuePair, keyValuePairNode.NodeType);

        var infixExpressionNode = parseNode.Children[1];
        Assert.AreEqual(FuncScriptParser.ParseNodeType.InfixExpression, infixExpressionNode.NodeType);
    }
}

File: ts/test/src/curated/nakedKeyValueCollectionWithImplicitReturn.test.ts
import { ParseContext, ParseNodeType } from "funcscript.ts.core/src/parser/FuncScriptParser.Main";
import { Parse } from "funcscript.ts.core/src/parser/syntax/FuncScriptParser.Parse";

test("NakedKeyValueCollectionWithImplicitReturn parsed correctly", () => {
  const expression = "x:12;x*2;";
  const c: ParseContext = {
    Expression: expression,
    SyntaxErrors: []
  };

  const res = Parse(c);

  expect(res.ParseNode).toBeDefined();
  expect(res.ParseNode!.NodeType).toBe(ParseNodeType.KeyValueCollection);
  expect(res.ParseNode!.Children.length).toBe(2);

  const keyValuePairNode = res.ParseNode!.Children[0];
  expect(keyValuePairNode.NodeType).toBe(ParseNodeType.KeyValuePair);

  const infixExpressionNode = res.ParseNode!.Children[1];
  expect(infixExpressionNode.NodeType).toBe(ParseNodeType.InfixExpression);
});